pkg/controller/sls/store_controller.go

```diff
@@ -0,0 +1,192 @@
+		}
+
+		pc := &v1alpha1.ProviderConfig{}
+		if err := c.client.Get(ctx, types.NamespacedName{Name: cr.Spec.ProviderConfigReference.Name}, pc); err != nil {
+			return nil, errors.Wrap(err, errGetProviderConfig)
+		}
+		if s := pc.Spec.Credentials.Source; s != xpv1.CredentialsSourceSecret {
+			return nil, errors.Errorf(errFmtUnsupportedCredSource, s)
+		}
+		sel = pc.Spec.Credentials.SecretRef
+		region = pc.Spec.Region
+	default:
+		return nil, errors.New(errNoProvider)
+	}
+
+	if sel == nil {
+		return nil, errors.New(errNoConnectionSecret)
+	}
+
+	s := &corev1.Secret{}
+	nn := types.NamespacedName{Namespace: sel.Namespace, Name: sel.Name}
+	if err := c.client.Get(ctx, nn, s); err != nil {
+		return nil, errors.Wrap(err, errGetConnectionSecret)
+	}
+
+	slsClient := c.NewClientFn(string(s.Data["accessKeyId"]), string(s.Data["accessKeySecret"]), region)
+	return &storeExternal{client: slsClient}, nil
+}
+
+type storeExternal struct {
+	client slsclient.LogClientInterface
+}
+
+func (e *storeExternal) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*slsv1alpha1.Store)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotStore)
+	}
+
+	storeName := meta.GetExternalName(cr)
+	project := cr.Spec.ForProvider.ProjectName
+
+	store, err := e.client.DescribeStore(project, storeName)
+	if slsclient.IsStoreNotFoundError(err) {
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	cr.Status.AtProvider = slsclient.GenerateStoreObservation(store)
+	upToDate := slsclient.IsStoreUpdateToDate(cr, store)
+	if upToDate {
+		cr.SetConditions(xpv1.Available())
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  upToDate,
+		ConnectionDetails: getStoreConnectionDetails(project, storeName),
+	}, nil
+}
+
+func (e *storeExternal) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*slsv1alpha1.Store)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotStore)
+	}
+	name := meta.GetExternalName(cr)
+	storeSpec := cr.Spec.ForProvider
+	cr.SetConditions(xpv1.Creating())
+	err := e.client.CreateStore(storeSpec.ProjectName, name, storeSpec.TTL, storeSpec.ShardCount, storeSpec.AutoSplit, storeSpec.MaxSplitShard)
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+	return managed.ExternalCreation{ConnectionDetails: getStoreConnectionDetails(storeSpec.ProjectName, name)}, nil
+}
+
+func (e *storeExternal) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*slsv1alpha1.Store)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotStore)
+	}
+	name := meta.GetExternalName(cr)
+	spec := cr.Spec.ForProvider
+	cr.Status.SetConditions(xpv1.Creating())
+	err := e.client.UpdateStore(spec.ProjectName, name, spec.TTL)
```

Comment:
nitpick, feel free to disregard if not your preferred pattern :)
```suggestion
	cr.Status.SetConditions(xpv1.Creating())
	err := e.client.UpdateStore(cr.Spec.ForProvider.ProjectName, meta.GetExternalName(cr), cr.Spec.ForProvider.TTL)
```